components/utils.js

```diff
@@ -0,0 +1,112 @@
+import useSWR from "swr";
+import { BsReverseLayoutTextWindowReverse } from "react-icons/bs";
+import { BsFilePlus } from "react-icons/bs";
+import { FaWrench } from "react-icons/fa";
+import { FaSearch } from "react-icons/fa";
+import { FaEdit } from "react-icons/fa";
+import { RiBankFill } from "react-icons/ri";
+import { FaDatabase } from "react-icons/fa";
+
+const fetcher = async (url) => {
+  const res = await fetch(url);
+  const text = await res.text();
+  if (!res.ok) {
+    throw text;
+  }
+  return JSON.parse(text);
+};
+
+export function useSocrata({ url }) {
+  const { data, error } = useSWR(url, fetcher);
+  return {
+    data: data || [],
+    isLoaded: !!error || !!data,
+    error: error,
+  };
+}
```

Comment:
here's the new data fetcher. the rest of this file contains functions extracted from elsewhere.